modules/core/src/test/java/org/apache/ignite/internal/processors/security/client/AdditionalSecurityCheckTest.java

```diff
@@ -152,4 +179,61 @@ public void testAdditionalPasswordServerFail() throws Exception {
 
         assertEquals(1, ignite.cluster().topologyVersion());
     }
+
+    /**
+     * Tests that the authentication request can be handled before a node is considered as completely started.
+     *
+     * @throws Exception If fails.
+     */
+    @Test
+    public void testAuthenticationBeforeNodeStart() throws Exception {
+        CountDownLatch startLatch = new CountDownLatch(1);
+        CountDownLatch contLatch = new CountDownLatch(1);
+        AtomicReference<Exception> err = new AtomicReference<>();
+
+        // Lifecycle bean that is used to block node start.
+        lifecycleBean = new LifecycleBean() {
+            /** Ignite instance. */
+            @IgniteInstanceResource
+            IgniteEx ignite;
+
+            /** {@inheritDoc} */
+            @Override public void onLifecycleEvent(LifecycleEventType evt) throws IgniteException {
+                if (evt == LifecycleEventType.BEFORE_NODE_START) {
+                    ignite.context().internalSubscriptionProcessor()
+                        .registerDistributedMetastorageListener(new DistributedMetastorageLifecycleListener() {
+                            @Override public void onReadyForRead(ReadableDistributedMetaStorage metastorage) {
+                                try {
+                                    startLatch.countDown();
+
+                                    if (contLatch.await(10, SECONDS))
+                                        throw new RuntimeException("Failed to wait for a latch to continue node start.");
+                                }
+                                catch (Exception e) {
+                                    err.set(e);
+                                }
+                            }
+                        });
+                }
+            }
+        };
+
+        IgniteInternalFuture<?> startFut = runAsync(() -> {
+            startGrid(0);
+        });
+
+        assertTrue("Failed to wait for starting node.", startLatch.await(10, SECONDS));
+
+        try (GridClient client = GridClientFactory.start(getGridClientConfiguration())) {
+            err.set(client.checkLastError());
+        }
+
+        contLatch.countDown();
+
+        startFut.get();
+
+        Exception unexpectedErr = err.get();
+
+        assertNull("Unexpected error [err=" + unexpectedErr + ']', unexpectedErr);
```

Comment:
```suggestion
        Exception unexpectedErr = err.get();

        assertNull(err.get());
```

Reply:
I don't like assertions without a message. On the other hand, in general, we should avoid reading err atomics twice. So, I would like to stay with the current code.